### PR TITLE
Fix sampling when `jaeger-baggage` header is given

### PR DESCRIPTION
**Pull request by yurishkuro**
Fix for #541

Signed: Yuri the Bug Maker

### CODE REVIEW

File: jaeger-core/src/test/java/io/jaegertracing/internal/ActiveSpanTest.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016, Uber Technologies, Inc
+ * Copyright (c) 2018, The Jaeger Authors.
```
Comment (black-adder) [inline jaeger-core/src/test/java/io/jaegertracing/internal/ActiveSpanTest.java]: `Copyright (c) 2016-2017, Uber Technologies, Inc`?

File: jaeger-core/src/test/java/io/jaegertracing/internal/ActiveSpanTest.java [hunk 2]
```diff
@@ -16,68 +16,50 @@
...
-    JaegerSpanContext jaegerSpanContext = tracer.extract(Format.Builtin.TEXT_MAP, carrier);
-    assertNotNull(jaegerSpanContext);
-    assertTrue(jaegerSpanContext.isDebugIdContainerOnly());
-    assertEquals("Coraline", jaegerSpanContext.getDebugId());
-
-    JaegerSpan span = tracer.buildSpan("span").asChildOf(jaegerSpanContext).start();
-    jaegerSpanContext = span.context();
-    assertTrue(jaegerSpanContext.isSampled());
-    assertTrue(jaegerSpanContext.isDebug());
-    assertEquals("Coraline", span.getTags().get(Constants.DEBUG_ID_HEADER_KEY));
```
Comment (yurishkuro) [inline jaeger-core/src/test/java/io/jaegertracing/internal/ActiveSpanTest.java]: moved to AdhocHeadersTest

File: jaeger-core/src/test/java/io/jaegertracing/internal/JaegerTracerTest.java [hunk 3]
```diff
@@ -167,13 +161,6 @@ public void testClose() {
     verify(sampler).close();
   }
 
-  @Test
-  public void testActiveSpan() {
-    JaegerSpan mockSpan = Mockito.mock(JaegerSpan.class);
-    tracer.scopeManager().activate(mockSpan, true);
-    assertEquals(mockSpan, tracer.activeSpan());
-  }
```
Comment (yurishkuro) [inline jaeger-core/src/test/java/io/jaegertracing/internal/JaegerTracerTest.java]: moved to ActiveSpanTest

File: jaeger-core/src/test/java/io/jaegertracing/internal/JaegerTracerTest.java [hunk 4]
```diff
@@ -185,42 +172,4 @@ public void testSpanContextNotSampled() {
...
-    // prepare
-    Span activeSpan = mock(Span.class);
-    ScopeManager scopeManager = tracer.scopeManager();
-
-    // test
-    scopeManager.activate(activeSpan, false);
-
-    // check
-    assertEquals(activeSpan, tracer.activeSpan());
-  }
```
Comment (yurishkuro) [inline jaeger-core/src/test/java/io/jaegertracing/internal/JaegerTracerTest.java]: moved to ActiveSpanTest

File: jaeger-core/src/test/java/io/jaegertracing/internal/JaegerTracerTest.java [hunk 5]
```diff
@@ -185,42 +172,4 @@ public void testSpanContextNotSampled() {
...
-
-  private void traceWithAdhocBaggage(Map<String, String> headers) {
-    headers.put("jaeger-baggage", "k1=v1, k2 = v2");
-
-    JaegerSpanContext parent = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapExtractAdapter(headers));
-    Span span = tracer.buildSpan("test").asChildOf(parent).start();
-
-    assertEquals("must have baggage", "v1", span.getBaggageItem("k1"));
-    assertEquals("must have baggage", "v2", span.getBaggageItem("k2"));
-  }
```
Comment (yurishkuro) [inline jaeger-core/src/test/java/io/jaegertracing/internal/JaegerTracerTest.java]: moved to AdhocHeadersTest

File: jaeger-core/src/test/java/io/jaegertracing/internal/ActiveSpanTest.java (hunk 1 above)
Comment (yurishkuro) [reply jaeger-core/src/test/java/io/jaegertracing/internal/ActiveSpanTest.java]: Nope, considering that active span did not exist in 2016 and most of 2017, and it was all implemented by people not from Uber.

**Pull request merged**